space_age.py

```diff
@@ -0,0 +1,29 @@
+class SpaceAge(object):
+    def __init__(self, seconds):
+        print('Earth = ' + '{0:.2f}'.format(seconds/60/60/24/365) + ' years old')
+        print('Mercury = ' + '{0:.2f}'.format(seconds/60/60/24/365*0.2408467) + ' years old')
+#why i get rounded integer not float/double figure for Earth
+
+print SpaceAge(71997456)
```

Comment:
@virtuallight i get strange message here, pls see last line from outcome:
Earth = 2.00 years old
Mercury = 0.48 years old
**<__main__.SpaceAge object at 0x7f76d39d7610>**

Reply:
`print SpaceAge(71997456)` means "print a representation of an object SpaceAge(71997456)". That object is of class `SpaceAge` which is a simple `object` type (as of definition `class SpaceAge(object)`). Those don't know how to represent themselves as a string. So they fall back to telling you the next best guess of what might be their string representation which is "<path.to.class.definition type at memory_address>". But changing that is most likely not what's being asked for in the exercise. Can you please link me to the exercise description or paste it here?

Reply:
> @zarniak Also, try running that against Python 3. It should complain about the "print" syntax 😏

@virtuallight changed compiler to https://www.onlinegdb.com/online_python_compiler and now is fine

Reply:
> `print SpaceAge(71997456)` means "print a representation of an object SpaceAge(71997456)". That object is of class `SpaceAge` which is a simple `object` type (as of definition `class SpaceAge(object)`). Those don't know how to represent themselves as a string. So they fall back to telling you the next best guess of what might be their string representation which is "<path.to.class.definition type at memory_address>". But changing that is most likely not what's being asked for in the exercise. Can you please link me to the exercise description or paste it here?

ok @virtuallight now i get it with print - i have changed the code only calling SpaceAge(71997456) (without print) and it is fine now

please find below text of exercise
"Given an age in seconds, calculate how old someone would be on:

Earth: orbital period 365.25 Earth days, or 31557600 seconds
Mercury: orbital period 0.2408467 Earth years
Venus: orbital period 0.61519726 Earth years
Mars: orbital period 1.8808158 Earth years
Jupiter: orbital period 11.862615 Earth years
Saturn: orbital period 29.447498 Earth years
Uranus: orbital period 84.016846 Earth years
Neptune: orbital period 164.79132 Earth years
So if you were told someone were 1,000,000,000 seconds old, you should be able to say that they're 31.69 Earth-years old.

If you're wondering why Pluto didn't make the cut, go watch this youtube video."

Reply:
@zarniak What do the tests look like for this task?

Reply:
> @zarniak What do the tests look like for this task?

@virtuallight 

there is something like .on_mercury() added to spaceage - what is it? how should i do that?

import unittest

from space_age import SpaceAge


# Tests adapted from `problem-specifications//canonical-data.json` @ v1.1.0

class SpaceAgeTest(unittest.TestCase):

    def test_age_on_mercury(self):
        self.assertEqual(SpaceAge(2134835688).on_mercury(), 280.88)

    def test_age_on_venus(self):
        self.assertEqual(SpaceAge(189839836).on_venus(), 9.78)

    def test_age_on_earth(self):
        self.assertEqual(SpaceAge(1000000000).on_earth(), 31.69)

    def test_age_on_mars(self):
        self.assertEqual(SpaceAge(2329871239).on_mars(), 39.25)

    def test_age_on_jupiter(self):
        self.assertEqual(SpaceAge(901876382).on_jupiter(), 2.41)

    def test_age_on_saturn(self):
        self.assertEqual(SpaceAge(3000000000).on_saturn(), 3.23)

    def test_age_on_uranus(self):
        self.assertEqual(SpaceAge(3210123456).on_uranus(), 1.21)

    def test_age_on_neptune(self):
        self.assertEqual(SpaceAge(8210123456).on_neptune(), 1.58)

    # Additional tests for this track

    def test_age_in_seconds(self):
        self.assertEqual(SpaceAge(1e6).seconds, 1e6)


if __name__ == '__main__':
    unittest.main()

Reply:
@zarniak Ok, so looking at the tests, what I think they are asking for is to define a class `SpaceAge` which accepts a single argument. Also, the `SpaceAge` class should implement several methods: `on_mercury`, `on_venus`, etc. Each of them should return a single value. There is an additional check "# Additional tests for this track" that hints that the object should store the number of provided seconds.

So, here is a rough scetch of what needs to be implemented:
```
class SpaceAge(object):
    __init__(self, seconds):
        self.seconds = seconds

    def on_mercury(self):
        # return age in Mercury-years
       
    def on_venus(self):
        # return age in Venus-years

    # etc.
```

Now, as for the implementation of each `on_...` method - I'd go for actually precalculating everything in the `__init__`. And then simply storing each result in fields like this:
```
    __init__(self, seconds):
        self.seconds = seconds
        # Provide proper calculations!
        self.mercury = 1.00
        self.venus = 2.00
        # etc.
```

Then return those values in corresponding methods.